File: 백준/1182/solution.py
import sys
N, S = map(int, input().split())
nums = list(map(int, input().split()))

def dfs(ind, total):
    global answer
    if total == S:
        answer += 1
    if ind >= len(nums):
        return
    for i in range(ind+1, len(nums)):
        dfs(i, total + nums[i])

answer = 0
for i in range(len(nums)):
    dfs(i, nums[i])
print(answer)

File: 백준/1182/solution.swift
import Foundation

// Input
let ns = readLine()!.split(separator: " ").compactMap{Int($0)}
let nums = readLine()!.split(separator: " ").compactMap{Int($0)}
let N = ns[0]
let S = ns[1]

// DFS
func dfs(ind : Int, total : Int) {
    if (total == S) {
        answer += 1
    }
    if (ind >= N) {
        return
    }
    for i in ind+1..<N {
        dfs(ind: i, total: total + nums[i])
    }
    
}

var answer = 0
for i in 0..<N {
    dfs(ind: i, total: nums[i])
}
print(answer)

